aws-cpp-sdk-core/include/aws/core/utils/threading/Executor.h

```diff
@@ -113,6 +113,45 @@ namespace Aws
 
                 friend class ThreadTask;
             };
+            
+            /**
+             * Executor extension that limits number of simultaneously running tasks by forcing
+             * additional tasks to wait if the number of running tasks reaches the specified maximum.
+             */
+            class AWS_CORE_API BlockingExecutor : public Executor
+            {
+            public:
+                /** Constructor that takes an Executor to delegate thread execution to and the max
+                 *  number of tasks that can run at once */
+                BlockingExecutor(std::shared_ptr<Executor> executor, size_t poolSize);
+                ~BlockingExecutor();
```

Comment:
since SubmitToThread is virtual, either mark it as final, or make the destructor virtual. My preference is for the virtual destructor.